src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -1231,7 +1231,31 @@ void ScaledDotProductAttention::initSupportedPrimitiveDescriptors() {
     config.outConfs[0].setMemDesc(
         creatorsMap.at(LayoutType::ncsp)->createSharedDesc(rtPrecision, getOutputShapeAtPort(0)));
 
-    supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::ref_any);
+    // Determine the correct implementation type based on compile flags and runtime capabilities
+    impl_desc_type implType = impl_desc_type::ref_any;
+#if defined(OPENVINO_ARCH_X86_64)
+    #ifdef OV_CPU_WITH_MLAS
+        // MLAS is preferred on x86 when available
+        implType = impl_desc_type::mlas;
```

Comment:
Looks like impl type [depends on](https://github.com/openvinotoolkit/openvino/blob/62c02eb1c1c7b6110fb05f982ae125cc03631f83/src/plugins/intel_cpu/src/nodes/scaled_attn.cpp#L1293) ISA and rtPrecision. If the hw doesn't support the rtPrecision by isa, the ref impl will be chosen. Or for example, I can see that fp32 kernel is [not supported](https://github.com/openvinotoolkit/openvino/blob/62c02eb1c1c7b6110fb05f982ae125cc03631f83/src/plugins/intel_cpu/src/nodes/scaled_attn.cpp#L1300) even on avx2 platforms.

Could you please double check?

Reply:
@a-sidorova I checked and fixed

Reply:
@a-sidorova I refactored implementation setting, please check